uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -139,14 +138,28 @@ protected function update($params) {
    */
   private function getErrorMessageForInvalidStatusTransition($params) {
     $leaveStatuses = LeaveRequest::getStatuses();
-    $isOwnRequest = CRM_Core_Session::getLoggedInContactID() === $params['contact_id'];
+    $isOwnRequest = $this->leaveRequestRightsService->currentUserIsLeaveContact($params['contact_id']);
 
     if ($isOwnRequest && (int)$params['status_id'] === (int)$leaveStatuses['approved']) {
-      return "You can't approve your own leave requests";
+      return 'You can\'t approve your own leave requests';
```

Comment:
Its fine to use double quotes here because of the can't. Would be more readable. Apologies, did't spot that earlier

Reply:
Np, done.